cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -1410,6 +1417,43 @@ TEST_F(TestLargeStringParquetIO, Basics) {
   this->RoundTripSingleColumn(large_array, large_array, arrow_properties);
 }
 
+using TestJsonParquetIO = TestParquetIO<::arrow::extension::JsonExtensionType>;
+
+TEST_F(TestJsonParquetIO, JsonExtension) {
+  const char* json = R"([
+    "null",
+    "1234",
+    "3.14159",
+    "true",
+    "false",
+    "\"a json string\"",
+    "[\"a\", \"json\", \"array\"]",
+    "{\"obj\": \"a simple json object\"}"
+  ])";
+
+  const auto json_type = ::arrow::extension::json();
+  const auto json_string_array = ::arrow::ArrayFromJSON(::arrow::utf8(), json);
+  const auto json_array = ::arrow::ExtensionType::WrapArray(json_type, json_string_array);
+
+  // When the original Arrow schema isn't stored and Arrow extensions are disabled,
+  // LogicalType::JSON is read as Binary.
+  const auto binary_array = ::arrow::ArrayFromJSON(::arrow::binary(), json);
```

Comment:
The PR description says

> This extension is backed by utf8()

So I would have naively expected this to be inferred as `arrow::utf8`?

Reply:
This is to preserve the existing behavior.

We currently read `LogicalType::JSON` as `arrow::binary`. So, if there is no schema information and the extensions are disabled, then we use the current behavior.

Reply:
Do you foresee an issue with my changing the Rust parquet reader to infer as UTF-8 in https://github.com/apache/arrow-rs/pull/3376, at least until such a time as this extension type is stabilised?

Reply:
I think that should be fine. The behavior should be implementation dependent. So, if the Rust implementation reads the Parquet JSON type as `utf8` it should continue to do that.

I'm not currently planning on touching the Rust implementation. If and when it is updated to support this extension, I would expect the current behavior to be preserved if the user requests the extension to be disabled.